xla/service/gpu/address_computation_fusion_rewriter.cc

```diff
@@ -409,15 +409,21 @@ absl::StatusOr<bool> AddressComputationFusionRewriter::Run(
                       std::pair<UseDefDataflowPaths, DefUseDataflowPaths>>
       matches;
 
-  // Collect all potential custom call matches in the non-fusion computations.
+  // Collect all potential custom call and reduce-scatter --> dus matches in the
+  // non-fusion computations.
   for (HloComputation* computation : module->computations()) {
     if (computation->IsFusionComputation()) continue;
     for (HloInstruction* instr : computation->instructions()) {
+      UseDefDataflowPaths sliced_operand_paths = {instr};
+      bool has_sliced_operand_paths = false;
       if (IsLegacyCublasMatmul(*instr) ||
           (IsCustomCall(instr, platform_name_))) {
-        UseDefDataflowPaths sliced_operand_paths = GetSlicedOperandPaths(instr);
-        bool has_sliced_operand_paths = sliced_operand_paths.size() > 1;
-
+        sliced_operand_paths = GetSlicedOperandPaths(instr);
+        has_sliced_operand_paths = sliced_operand_paths.size() > 1;
+      }
+      if (instr->opcode() == HloOpcode::kReduceScatter ||
+          IsLegacyCublasMatmul(*instr) ||
+          (IsCustomCall(instr, platform_name_))) {
```

Comment:
Minor nit: do you need the parens outside the `IsCustomCall` call?

Reply:
I dont think so. I will edit this, and change this in the condition on line 417 too.